specification/resources/resource-manager/Microsoft.Resources/stable/2024-04-01/resources.json

```diff
@@ -0,0 +1,6839 @@
+            "type": "string",
+            "description": "The additional properties. "
+          },
+          "description": "The properties."
+        }
+      },
+      "description": "Resource type managed by the resource provider."
+    },
+    "Provider": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provider ID."
+        },
+        "namespace": {
+          "type": "string",
+          "description": "The namespace of the resource provider."
+        },
+        "registrationState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The registration state of the resource provider."
+        },
+        "registrationPolicy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The registration policy of the resource provider."
+        },
+        "resourceTypes": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ProviderResourceType"
+          },
+          "x-ms-identifiers": [
+            "resourceType"
+          ],
+          "description": "The collection of provider resource types."
+        },
+        "providerAuthorizationConsentState": {
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Required",
+            "NotRequired",
+            "Consented"
+          ],
+          "description": "The provider authorization consent state.",
+          "x-ms-enum": {
+            "name": "ProviderAuthorizationConsentState",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "Resource provider information."
+    },
+    "BasicDependency": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the dependency."
+        },
+        "resourceType": {
+          "type": "string",
+          "description": "The dependency resource type."
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "The dependency resource name."
+        }
+      },
+      "description": "Deployment dependency information."
+    },
+    "Dependency": {
+      "properties": {
+        "dependsOn": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BasicDependency"
+          },
+          "description": "The list of dependencies."
+        },
+        "id": {
+          "type": "string",
+          "description": "The ID of the dependency."
+        },
+        "resourceType": {
+          "type": "string",
+          "description": "The dependency resource type."
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "The dependency resource name."
+        }
+      },
+      "description": "Deployment dependency information."
+    },
+    "PreflightWarningResponseInfo": {
```

Comment:
[nit] rename to something not preflight-specific (`DeploymentWarning`)?

Reply:
Initially this was scoped to Preflight related warnings, we can extend this to Preflight + Deployments, and have warning types for Preflight for example we can have -> NestedDeploymentExpansionWarnings, ResourceRegistrationWarnings, and for Deployments same thing.

Reply:
[nit] How about just `DeploymentDiagnostics`? The `Definition` part feels unnecessary.